### PR TITLE
Quick Start for Existing Users: Add new tasks collection for existing users

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #18388

## Description
- Added feature flag
- Added the new "Get to know the WP app" collection
- Display the new collection for existing users and the old ones otherwise.
- Saved type of the quick start in progress (new site/existing site) in `Blog`

PS: I added an item in debug menu to enable Quick Start for existing sites

Expaned View | Card
-|-
Simulator Screen Shot - iPhone 13 Pro - 2022-04-21 at 07 46 25|Simulator Screen Shot - iPhone 13 Pro - 2022-04-21 at 07 46 23

### Login
1. Install the app
2. Logout if needed
3. Login using an existing account
4. Choose a site to open
5. Tap "Show me around"
6. Make sure that the new set of tasks is displayed in the quick start card
7. Open the extended quick start view
8. Make sure the title displayed is correct
9. Go through the steps and make sure the card and extended view are updated correctly.
10. In the middle of the tour, terminate the app and launch it again and check that progress is persisted 

### Signup
1. Install the app
2. Logout if needed
3. Sign up for a new account
4. Create a site
5. Tap "Show me around"
6. Make sure that the old set of tasks is displayed in the quick start card
7. Do the same smoke test described above

### Site creation after login
1. Install the app
2. Logout if needed
3. Login using an existing account
4. Tap "Create a new site"
5. Create a site
6. Tap "Show me around"
7. Make sure that the old set of tasks is displayed in the quick start card

### Site creation while logged in
1. Install the app
2. Make sure to be logged in
3. Tap the site picker
4. Tap "+" button
5. Create a site
6. Tap "Show me around"
7. Make sure that the old set of tasks is displayed in the quick start card

### Removing Quick Start
1. Enable quick start for a site (any type)
2. Tap the ellipsis button and remove quick start
3. Make sure quick start gets removed normally
4. Relaunch the app
5. Make sure quick start is not displayed

### Opening site after logic with feature flag disabled
1. Manually disable feature flag from code
2. Install the app
3. Logout if needed
4. Login using an existing account
5. Choose a site to open
6. Tap "Show me around"
7. Make sure that the old set of tasks is displayed in the quick start card

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/QuickStartToursCollection.swift [hunk 1]
```diff
@@ -59,3 +59,25 @@ struct QuickStartGrowToursCollection: QuickStartToursCollection {
...
+    let completedImageName: String
+    let analyticsKey: String
+    let tours: [QuickStartTour]
+
+    init(blog: Blog) {
+        self.title = NSLocalizedString("Get to know the WordPress app",
+                                      comment: "Name of the Quick Start list that guides users through a few tasks to explore the WordPress app.")
+        self.hint = NSLocalizedString("A series of steps helping you to explore the app.",
+                                     comment: "A VoiceOver hint to explain what the user gets when they select the 'Get to know the WordPress app' button.")
+        self.completedImageName = "wp-illustration-tasks-complete-site" // TODO: Could be changed
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/Blog/QuickStartToursCollection.swift]: Ref: pbArwn-4kz-p2#comment-5556

File: WordPress/Classes/ViewRelated/Me/App Settings/DebugMenuViewController.swift [hunk 2]
```diff
@@ -68,8 +68,11 @@ class DebugMenuViewController: UITableViewController {
 
     private var toolsRows: [ImmuTableRow] {
         var toolsRows = [
-            ButtonRow(title: Strings.quickStartRow, action: { [weak self] _ in
-                self?.displayBlogPickerForQuickStart()
+            ButtonRow(title: Strings.quickStartForNewSiteRow, action: { [weak self] _ in
+                self?.displayBlogPickerForQuickStart(type: .newSite)
+            }),
+            ButtonRow(title: Strings.quickStartForExistingSiteRow, action: { [weak self] _ in
+                self?.displayBlogPickerForQuickStart(type: .existingSite)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Me/App Settings/DebugMenuViewController.swift:75]: 😍

File: WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift [hunk 3]
```diff
@@ -575,7 +575,8 @@ private extension WordPressAuthenticationManager {
         // Otherwise, show the My Site screen for the specified blog and after a short delay,
         // trigger the Quick Start tour
         self.windowManager.showAppUI(for: blog, completion: {
-            QuickStartTourGuide.shared.setupWithDelay(for: blog)
+            let quickStartType: QuickStartType = FeatureFlag.quickStartForExistingUsers.enabled ? .existingSite : .newSite
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift]: RE: Existing Site tasks shown for the **Site creation after login** flow 

This is where the `.existingSite` type is assigned

**Review by momo-ozawa — COMMENTED** [review]
@hassaanelgarem 

For the **Site creation after login** flow, I'm getting the new QS tasks instead of the old ones. All the other flows work as described.

**Comment by hassaanelgarem** [comment]
@momo-ozawa 

> For the Site creation after login flow, I'm getting the new QS tasks instead of the old ones.

Fixed in 9f90704332d2e390151da3ce9be1a3fe8b5d9d4c

I also forgot this test case:
### Opening site after login with feature flag disabled
1. Manually disable feature flag from code
2. Install the app
3. Logout if needed
4. Login using an existing account
5. Choose a site to open
6. Tap "Show me around"
7. Make sure that the old set of tasks is displayed in the quick start card

Could you please check it as well? 🙏

File: WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift (hunk 3 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift]: Updated in 9f90704332d2e390151da3ce9be1a3fe8b5d9d4c 👍

File: WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift [hunk 4]
```diff
@@ -335,7 +335,7 @@ extension WordPressAuthenticationManager: WordPressAuthenticatorDelegate {
         // If adding a self-hosted site, skip the Epilogue
         if let wporg = credentials.wporg,
            let blog = Blog.lookup(username: wporg.username, xmlrpc: wporg.xmlrpc, in: ContextManager.shared.mainContext) {
-            presentQuickStartPrompt(for: blog, in: navigationController, onDismiss: onDismissQuickStartPromptForNewSiteHandler)
+            presentQuickStartPrompt(for: blog, in: navigationController, onDismiss: onDismissQuickStartPromptForNewSiteHandler) // TODO: Could be changed
```
Comment (hassaanelgarem) [inline WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift]: Ref: p1650558436924469/1650522675.714159-slack-C027K4MNPGQ

**Review by momo-ozawa — APPROVED** [review]
Tested these flows and it works as described!
- Opening site after login with feature flag disabled
- Site creation after login

**Pull request merged**